src/dicomweb_client/uri.py

```diff
@@ -0,0 +1,358 @@
+"""Utilities for DICOMweb URI manipulation."""
+import enum
+import re
+from typing import Optional, Tuple
+import urllib.parse as urlparse
+
+
+class URIType(enum.Enum):
```

Comment:
I just realized that `frame` is missing. A URI may have the following pattern

``<base_url>/studies/<study_instance_uid>/series/<series_instance_uid>/instances/<sop_instance_uid>/frames/<frame_numbers>``

Reply:
Do we want to add type `FRAME = 'frame'`?

Reply:
This would require changes to the API of `URI`. At a minimum, `URI.from_string()` and `URI.update()` should consider and keep track of any suffixes (see comment below).

Reply:
Sure thing!